crates/relayer/src/event/monitor.rs

```diff
@@ -176,14 +176,19 @@ impl EventMonitor {
     pub fn new(
         chain_id: ChainId,
         node_addr: Url,
+        rpc_compat: CompatMode,
```

Comment:
@romac There's also [version discovery code](
https://github.com/informalsystems/hermes/blob/0e81ee6bdaccaf44c407151d58589ab055c95657/crates/relayer/src/chain/cosmos.rs#L718) when instantiating the inner RPC client. That can be removed now and the HTTP client initialized with the previously discovered compatibility mode. On the other hand, it would be still better to reuse that HTTP client's connection to discover the version for the websocket client, rather than connecting twice in a row.